extra/linux/redhat/alacritty.spec

```diff
@@ -7,14 +7,16 @@ URL:           https://github.com/alacritty/alacritty
 VCS:           https://github.com/alacritty/alacritty.git
 Source:        alacritty-%{version}.tar
 
-BuildRequires: rust >= 1.41.0
+BuildRequires: rust >= 1.43.0
 BuildRequires: cargo
 BuildRequires: cmake
-BuildRequires: freetype-devel
-BuildRequires: fontconfig-devel
-BuildRequires: libxcb-devel
+BuildRequires: gcc-c++
+BuildRequires: python3
+BuildRequires: pkgconfig(freetype2)
+BuildRequires: pkgconfig(fontconfig)
+BuildRequires: pkgconfig(xcb)
```

Comment:
What does this `pkgconfig` annotation mean? The `-devel` seems like the normal way to show dev dependencies.

Reply:
The library has provided a pkg-config file (pc) that can is used by rpm to meet the dependency. 
  
Fedora recommends for libraries that use pkg-config that the build dependency be done that way.

ref[1]: https://docs.fedoraproject.org/en-US/packaging-guidelines/PkgConfigBuildRequires/
ref[2]: https://people.freedesktop.org/~dbn/pkg-config-guide.html
ref[3]: https://rusnak.io/why-is-pkg-config-the-best-thing-since-sliced-bread/

Reply:
How do you know that these are all using pkgconfig? I'm not sure that `xcb` is...

Reply:
It can be verified with DNF, for libxcb-devel: https://paste.centos.org/view/206ab29

Thanks

Reply:
You're saying that these are the pkgconfig things that pull libxcb-devel, right? But I'm saying that I'm not sure if the xcb crate uses pkgconfig. In fact I don't think it is.

Reply:
> You're saying that these are the pkgconfig things that pull libxcb-devel, right? But I'm saying that I'm not sure if the xcb crate uses pkgconfig. In fact I don't think it is.

BuildRequires tells rpmbuild what is needed at build time.
Requires is what's needed at runtime. 

It's not for the crate, it's to meet the build dependencies.

Reply:
I'm aware. I'm just saying that the `xcb` crate is not using `pkgconfig` to link against dependencies at build time. The Freetype/Fontconfig stuff is, but xcb is not.

I'm fairly certain that `xcb` can be compiled without `pkgconfig` installed at all, thus specifying it as a dependency that links against `xcb` using `pkgconfig` is false.

Reply:
Switched back to -devel, thanks.